tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,321 @@
+    expect(stdout).toContain('LIVE');
+    await sleep(1100);
+    const token2 = await session.readToken();
+    expect(token1).not.toBe(token2);
   });
-
-  describe('After session has been unlocked', () => {
-    beforeAll(async () => {
-      // Authorize session
-      await testUtils.pkStdio(
-        ['agent', 'unlock', '-np', nodePath, '--password-file', passwordFile],
-        {},
-        dataDir,
-      );
-    }, global.polykeyStartupTimeout);
-
-    test('Process should refresh the session token', async () => {
-      tokenBuffer = await fs.promises.readFile(sessionFile);
-      token = tokenBuffer.toString() as SessionToken;
-      const prevToken = token;
-
-      // At least 1 second of delay
-      // Expiry time resolution is in seconds
-      await sleep(1100);
-      const result = await testUtils.pkStdio(command, {}, dataDir);
-      expect(result.exitCode).toBe(0);
-      expect(result.stdout).toContain(vaultName);
-
-      const buff = await fs.promises.readFile(sessionFile);
-      const newToken = buff.toString() as SessionToken;
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(newToken),
-      ).not.toThrow();
-      // Make sure the first and second tokens are not the same
-      expect(newToken).not.toEqual(prevToken);
+  test('serial processes should both refresh the session token', async () => {
+    const session = await Session.createSession({
+      sessionTokenPath,
+      fs,
+      logger,
     });
-
-    test('Serial processes should refresh the session token', async () => {
-      let result = await testUtils.pkStdio(command, {}, dataDir);
-      expect(result.exitCode).toBe(0);
-      expect(result.stdout).toContain(vaultName);
-
-      tokenBuffer = await fs.promises.readFile(sessionFile);
-      const token1 = tokenBuffer.toString() as SessionToken;
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(token1),
-      ).not.toThrow();
-
-      // At least 1 second of delay
-      // Expiry time resolution is in seconds
-      await sleep(1100);
-      result = await testUtils.pkStdio(command, {}, dataDir);
-      expect(result.exitCode).toBe(0);
-      expect(result.stdout).toContain(vaultName);
-
-      tokenBuffer = await fs.promises.readFile(sessionFile);
-      const token2 = tokenBuffer.toString() as SessionToken;
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(token2),
-      ).not.toThrow();
-
-      // Make sure the first and second tokens are not the same
-      expect(token1).not.toEqual(token2);
+    // Run first command
+    let exitCode, stdout;
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    ));
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
+    await sleep(1100);
```

Comment:
Remove all of these sleeps.

Reply:
All sleeps have been removed in favour of using `pkExec` (to enforce the asynchronous session token write completes at conclusion of command) with the newly added typescript-cached-transpile (to improve execution time of the tests).

There's one residual sleep in `failing processes should unlock the session file`, but this test will potentially be removed, as per https://github.com/MatrixAI/js-polykey/pull/296#discussion_r765357879